core/domain/role_services.py

```diff
@@ -122,211 +123,142 @@
+    ACTION_CREATE_EXPLORATION,
+    ACTION_DELETE_OWNED_PRIVATE_ACTIVITY,
+    ACTION_EDIT_OWNED_ACTIVITY,
+    ACTION_SUBSCRIBE_TO_USERS,
+    ACTION_MANAGE_ACCOUNT,
+    ACTION_MODIFY_ROLES_FOR_OWNED_ACTIVITY,
+    ACTION_PUBLISH_OWNED_ACTIVITY,
+    ACTION_RATE_ANY_PUBLIC_EXPLORATION,
+    ACTION_SUGGEST_CHANGES,
+    ACTION_SUBMIT_VOICEOVER_APPLICATION,
+    *_LEARNER_ALLOWED_ACTIONS)
+
+_COLLECTION_EDITOR_ALLOWED_ACTIONS = _get_unique_actions_list(
+    ACTION_CREATE_COLLECTION,
+    *_EXPLORATION_EDITOR_ALLOWED_ACTIONS)
+
+_TOPIC_MANAGER_ALLOWED_ACTIONS = _get_unique_actions_list(
+    ACTION_ACCESS_TOPICS_AND_SKILLS_DASHBOARD,
+    ACTION_DELETE_ANY_QUESTION,
+    ACTION_EDIT_ANY_QUESTION,
+    ACTION_EDIT_OWNED_STORY,
+    ACTION_EDIT_OWNED_TOPIC,
+    ACTION_EDIT_SKILLS,
+    ACTION_EDIT_ANY_SUBTOPIC_PAGE,
+    ACTION_MANAGE_QUESTION_SKILL_STATUS,
+    ACTION_VISIT_ANY_QUESTION_EDITOR,
+    ACTION_VISIT_ANY_TOPIC_EDITOR,
+    *_COLLECTION_EDITOR_ALLOWED_ACTIONS)
+
+_MODERATOR_ALLOWED_ACTIONS = _get_unique_actions_list(
+    ACTION_ACCESS_MODERATOR_PAGE,
+    ACTION_DELETE_ANY_PUBLIC_ACTIVITY,
+    ACTION_EDIT_ANY_PUBLIC_ACTIVITY,
+    ACTION_PLAY_ANY_PRIVATE_ACTIVITY,
+    ACTION_SEND_MODERATOR_EMAILS,
+    ACTION_UNPUBLISH_ANY_PUBLIC_ACTIVITY,
+    *_TOPIC_MANAGER_ALLOWED_ACTIONS)
+
+_ADMIN_ALLOWED_ACTIONS = _get_unique_actions_list(
+    ACTION_ACCEPT_ANY_SUGGESTION,
+    ACTION_ACCEPT_ANY_VOICEOVER_APPLICATION,
+    ACTION_CHANGE_STORY_STATUS,
+    ACTION_CHANGE_TOPIC_STATUS,
+    ACTION_CREATE_NEW_SKILL,
+    ACTION_CREATE_NEW_TOPIC,
+    ACTION_DELETE_ANY_ACTIVITY,
+    ACTION_DELETE_ANY_SKILL,
+    ACTION_DELETE_TOPIC,
+    ACTION_EDIT_ANY_ACTIVITY,
+    ACTION_EDIT_ANY_STORY,
+    ACTION_EDIT_ANY_TOPIC,
+    ACTION_EDIT_SKILLS,
+    ACTION_EDIT_SKILL_DESCRIPTION,
+    ACTION_MANAGE_EMAIL_DASHBOARD,
+    ACTION_MANAGE_TOPIC_RIGHTS,
+    ACTION_MODIFY_ROLES_FOR_ANY_ACTIVITY,
+    ACTION_PUBLISH_ANY_ACTIVITY,
+    ACTION_PUBLISH_OWNED_SKILL,
+    *_MODERATOR_ALLOWED_ACTIONS)
+
+# This dict represents all the actions that belong to a particular role.
+_ROLE_ACTIONS = {
+    feconf.ROLE_ID_ADMIN: _ADMIN_ALLOWED_ACTIONS,
+    feconf.ROLE_ID_BANNED_USER: [],
+    feconf.ROLE_ID_COLLECTION_EDITOR: _COLLECTION_EDITOR_ALLOWED_ACTIONS,
+    feconf.ROLE_ID_EXPLORATION_EDITOR: _EXPLORATION_EDITOR_ALLOWED_ACTIONS,
+    feconf.ROLE_ID_GUEST: _GUEST_ALLOWED_ACTIONS,
+    feconf.ROLE_ID_LEARNER: _LEARNER_ALLOWED_ACTIONS,
+    feconf.ROLE_ID_MODERATOR: _MODERATOR_ALLOWED_ACTIONS,
+    feconf.ROLE_ID_TOPIC_MANAGER: _TOPIC_MANAGER_ALLOWED_ACTIONS
 }
 
 
 def get_all_actions(role):
-    """Returns a list of all actions (including inherited actions)
-    that can be performed by the given role.
+    """Returns a list of all actions that can be performed by the given role.
 
     Args:
-        role: str. A string defining user role. It should be a key of
-            PARENT_ROLES.
+        role: str. A string defining user role.
```

Comment:
defining user role --> defining the user role

Reply:
DOne!